readiness-tool/getPagesSource.js

```diff
@@ -0,0 +1,32 @@
+function DOMtoString(document_root) {
+    var html = '',
+        node = document_root.firstChild;
+    while (node) {
+        switch (node.nodeType) {
+        case Node.ELEMENT_NODE:
+            html += node.outerHTML;
+            break;
+        case Node.TEXT_NODE:
+            html += node.nodeValue;
+            break;
+        case Node.CDATA_SECTION_NODE:
+            html += '<![CDATA[' + node.nodeValue + ']]>';
+            break;
+        case Node.COMMENT_NODE:
+            html += '<!--' + node.nodeValue + '-->';
+            break;
+        case Node.DOCUMENT_TYPE_NODE:
+            // (X)HTML documents are identified by public identifiers
+            html += "<!DOCTYPE " + node.name + (node.publicId ? ' PUBLIC "' + node.publicId + '"' : '') + (!node.publicId && node.systemId ? ' SYSTEM' : '') + (node.systemId ? ' "' + node.systemId + '"' : '') + '>\n';
```

Comment:
Please keep lines to 80 chars max. Depending on your editor, you can add a ruler to show when a line goes over.